lib/main.js

```diff
@@ -1,156 +1,87 @@
-'use babel';
+const path = require('path');
+const { AutoLanguageClient } = require('atom-languageclient');
 
-// eslint-disable-next-line import/extensions, import/no-extraneous-dependencies
-import { CompositeDisposable, Task } from 'atom';
-import path from 'path';
-import fs from 'fs';
-import * as workerHelper from './workerHelper';
+class TSLintLanguageClient extends AutoLanguageClient {
+  getGrammarScopes() {
+    return atom.config.get('linter-tslint.javascriptSupport')
```

Comment:
Settings should always be observed, where would the best place to do that be here? `startServerProcess`? Or maybe in the constructor?

Reply:
Ideally they would but it gets complex fast - you'd need to handle open files of the type supported/no longer supported and send the appropriate didopen or didclose events.  It's a lot of complexity for uncommon scenario.  I'm also not sure that you can force many of the providers to re-evaluate the service endpoints that passed back these grammar scopes.

You might just be better off removing it - it was a last-minute workaround for people who need flow and typescript installed side-by-side.

Reply:
Oh, I was talking about in general, there are far more uses of `atom.config.get` in this code.

Reply:
Taken care of!